components/application-connectivity-validator/internal/validationproxy/handler.go

```diff
@@ -113,22 +153,36 @@ func hasValidSubject(subjects []string, appName, group, tenant string) bool {
 	return false
 }
 
-func newSubjectValidator(appName, group, tenant string) func(subject pkix.Name) bool {
-	validateCommonName := func(subject pkix.Name) bool {
+func newSubjectValidator(applicationClientIDs []string, appName, group, tenant string) func(subject pkix.Name) bool {
+	validateCommonNameWithAppName := func(subject pkix.Name) bool {
 		return appName == subject.CommonName
 	}
-
-	if group == "" || tenant == "" {
-		return validateCommonName
+	validateCommonNameWithClientIDs := func(subject pkix.Name) bool {
+		for _, id := range applicationClientIDs {
+			if subject.CommonName == id {
+				return true
+			}
+		}
+		return false
 	}
-
-	return func(subject pkix.Name) bool {
-		return validateCommonName(subject) && validateSubjectField(subject.Organization, tenant) && validateSubjectField(subject.OrganizationalUnit, group)
+	validateSubjectField := func(subjectField []string, expectedValue string) bool {
+		return len(subjectField) == 1 && subjectField[0] == expectedValue
 	}
-}
 
-func validateSubjectField(subjectField []string, expectedValue string) bool {
-	return len(subjectField) == 1 && subjectField[0] == expectedValue
+	switch {
+	case group == "" || tenant == "":
+		return validateCommonNameWithAppName
+
```

Comment:
As discussed we should adjust this logic.

Reply:
Done!